Userland/Utilities/unzip.cpp

```diff
@@ -17,6 +17,33 @@
 #include <LibCrypto/Checksum/CRC32.h>
 #include <sys/stat.h>
 
+static time_t dos_to_time_t(u16 dos_time, u16 dos_date)
+{
+    struct tm tm;
+
+    tm.tm_isdst = -1;
+    tm.tm_year = ((dos_date >> 9) & 127) + 1980 - 1900;
+    tm.tm_mon = ((dos_date >> 5) & 15) - 1;
+    tm.tm_mday = dos_date & 31;
+
+    tm.tm_hour = (dos_time >> 11) & 31;
+    tm.tm_min = (dos_time >> 5) & 63;
+    tm.tm_sec = (dos_time << 1) & 62;
+
+    return mktime(&tm);
+}
```

Comment:
This should be somewhere inside `LibArchive`. It also reminds me a lot of the time translation in our FAT filesystem implementation, maybe there is some opportunity to share code.

Reply:
It actually works exactly the same like the `FATPacked*`  times. I oversaw this unfortunately. Therefore the code can be shared very well.
Do you have any recommendation of how to do this in the best way ? Should I create a class called e.g. `MSPackedDate` (`MS` = Microsoft) that handles the translations?

Reply:
The more interesting question is _where_ to put it. I thought of just adding conversion functions to `AK::Time`, but that is apparently for durations, not for an absolute date-time. Maybe we should just create `AK::DateTime` to just have _somewhere_ to put it and look what others think about it.

Reply:
Eh, looks like `AK::Time` _does_ support absolute datetimes, so we would just add conversion methods to/from the packed DOS format there.

Reply:
Problem with `AK::Time` is that is has no support to translate an absolute datetime back to a date

Reply:
Decided to make a dedicated file and host the conversion functions there since they didn't seem to fit nicely into `AK/Time` since it doesn't support conversion from absolute times to date.

If this approach is alright, I would continue to refactor the FAT code to use the conversion functions as well.